#### Part 1 - uncleaned data

In [1]:
# Dependencies and Setup
import matplotlib.pyplot as plt
import pandas as pd
import scipy.stats as st
import numpy as np

#### Question 1:

#### How many total reservations are in this dataset?

92,545

In [2]:
# Import the data/csv file & preview
original = pd.read_csv("vacasa_data.csv")
df = pd.read_csv("vacasa_data.csv")
test = pd.read_csv("vacasa_data.csv")
df

unit_id check_in_date check_out_date reservation_type  \
0        40630      1/1/2018       1/3/2018            guest   
1        40630      1/3/2018       1/7/2018            guest   
2        40630     1/11/2018      1/15/2018            guest   
3        40630     1/15/2018      1/18/2018            owner   
4        40630     1/20/2018      1/21/2018            guest   
...        ...           ...            ...              ...   
92540    21239      4/9/2019      4/12/2019            guest   
92541    21239     4/12/2019      4/14/2019            guest   
92542    21239     4/14/2019      4/16/2019            guest   
92543    21239     4/16/2019      5/31/2019            owner   
92544    21239     5/31/2019     12/31/2021            owner   

             reservation_source_type reservation_created_date  adult_guests  \
0                            unknown                 1/1/2018             2   
1                            unknown                 1/3/2018             2   
2                            unknown                9/29/2017             2   
3                            unknown                1/15/2018             2   
4                            unknown                1/17/2018             2   
...                              ...                      ...           ...   
92540  3rd Party Channel Reservation                 4/8/2019             2   
92541                         Vacasa                 4/8/2019             2   
92542  3rd Party Channel Reservation                 4/8/2019             6   
92543                         Vacasa                4/13/2019             0   
92544                         Vacasa                4/13/2019             0   

       child_guests  pets total_paid  city_name   zip5 state_name_short  \
0                 0     0     $1,930       Vail  81657               CO   
1                 0     0     $2,969       Vail  81657               CO   
2                 0     0    $12,913       Vail  81657               CO   
3                 0     0        $60       Vail  81657               CO   
4                 0     0       $515       Vail  81657               CO   
...             ...   ...        ...        ...    ...              ...   
92540             0     0       $340  Galveston  77554               TX   
92541             1     0       $245  Galveston  77554               TX   
92542             0     0       $287  Galveston  77554               TX   
92543             0     0         $0  Galveston  77554               TX   
92544             0     0         $0  Galveston  77554               TX   

          region_name_short  
0       Vail / Beaver Creek  
1       Vail / Beaver Creek  
2       Vail / Beaver Creek  
3       Vail / Beaver Creek  
4       Vail / Beaver Creek  
...                     ...  
92540  The Texas Gulf Coast  
92541  The Texas Gulf Coast  
92542  The Texas Gulf Coast  
92543  The Texas Gulf Coast  
92544  The Texas Gulf Coast  

[92545 rows x 14 columns]

#### Question 2:

#### What's the number of units that had a reservation?

2,197

In [3]:
# Find all reservations 
df['unit_id'].value_counts()

9131     177
11033    163
12006    161
10143    159
9390     159
        ... 
42492      1
62902      1
18488      1
43032      1
42445      1
Name: unit_id, Length: 2197, dtype: int64

#### This is the start of where I cleaned the data, for the other notebook file. 

In [4]:
# Remove $ and , from total_paid
df['total_paid'] = df['total_paid'].str.replace('$', '')
df['total_paid'] = df['total_paid'].str.replace(',', '')
df


unit_id check_in_date check_out_date reservation_type  \
0        40630      1/1/2018       1/3/2018            guest   
1        40630      1/3/2018       1/7/2018            guest   
2        40630     1/11/2018      1/15/2018            guest   
3        40630     1/15/2018      1/18/2018            owner   
4        40630     1/20/2018      1/21/2018            guest   
...        ...           ...            ...              ...   
92540    21239      4/9/2019      4/12/2019            guest   
92541    21239     4/12/2019      4/14/2019            guest   
92542    21239     4/14/2019      4/16/2019            guest   
92543    21239     4/16/2019      5/31/2019            owner   
92544    21239     5/31/2019     12/31/2021            owner   

             reservation_source_type reservation_created_date  adult_guests  \
0                            unknown                 1/1/2018             2   
1                            unknown                 1/3/2018             2   
2                            unknown                9/29/2017             2   
3                            unknown                1/15/2018             2   
4                            unknown                1/17/2018             2   
...                              ...                      ...           ...   
92540  3rd Party Channel Reservation                 4/8/2019             2   
92541                         Vacasa                 4/8/2019             2   
92542  3rd Party Channel Reservation                 4/8/2019             6   
92543                         Vacasa                4/13/2019             0   
92544                         Vacasa                4/13/2019             0   

       child_guests  pets total_paid  city_name   zip5 state_name_short  \
0                 0     0       1930       Vail  81657               CO   
1                 0     0       2969       Vail  81657               CO   
2                 0     0      12913       Vail  81657               CO   
3                 0     0         60       Vail  81657               CO   
4                 0     0        515       Vail  81657               CO   
...             ...   ...        ...        ...    ...              ...   
92540             0     0        340  Galveston  77554               TX   
92541             1     0        245  Galveston  77554               TX   
92542             0     0        287  Galveston  77554               TX   
92543             0     0          0  Galveston  77554               TX   
92544             0     0          0  Galveston  77554               TX   

          region_name_short  
0       Vail / Beaver Creek  
1       Vail / Beaver Creek  
2       Vail / Beaver Creek  
3       Vail / Beaver Creek  
4       Vail / Beaver Creek  
...                     ...  
92540  The Texas Gulf Coast  
92541  The Texas Gulf Coast  
92542  The Texas Gulf Coast  
92543  The Texas Gulf Coast  
92544  The Texas Gulf Coast  

[92545 rows x 14 columns]

In [5]:
# Remove rows that have $0 sales
df.drop(df.loc[df['total_paid']== "0"].index, inplace=True)
df


unit_id check_in_date check_out_date reservation_type  \
0        40630      1/1/2018       1/3/2018            guest   
1        40630      1/3/2018       1/7/2018            guest   
2        40630     1/11/2018      1/15/2018            guest   
3        40630     1/15/2018      1/18/2018            owner   
4        40630     1/20/2018      1/21/2018            guest   
...        ...           ...            ...              ...   
92536    21239      3/9/2019      3/13/2019            guest   
92538    21239     3/21/2019      3/24/2019            guest   
92540    21239      4/9/2019      4/12/2019            guest   
92541    21239     4/12/2019      4/14/2019            guest   
92542    21239     4/14/2019      4/16/2019            guest   

             reservation_source_type reservation_created_date  adult_guests  \
0                            unknown                 1/1/2018             2   
1                            unknown                 1/3/2018             2   
2                            unknown                9/29/2017             2   
3                            unknown                1/15/2018             2   
4                            unknown                1/17/2018             2   
...                              ...                      ...           ...   
92536  3rd Party Channel Reservation                2/17/2019             4   
92538  3rd Party Channel Reservation                3/20/2019             4   
92540  3rd Party Channel Reservation                 4/8/2019             2   
92541                         Vacasa                 4/8/2019             2   
92542  3rd Party Channel Reservation                 4/8/2019             6   

       child_guests  pets total_paid  city_name   zip5 state_name_short  \
0                 0     0       1930       Vail  81657               CO   
1                 0     0       2969       Vail  81657               CO   
2                 0     0      12913       Vail  81657               CO   
3                 0     0         60       Vail  81657               CO   
4                 0     0        515       Vail  81657               CO   
...             ...   ...        ...        ...    ...              ...   
92536             0     0       1300  Galveston  77554               TX   
92538             0     0        661  Galveston  77554               TX   
92540             0     0        340  Galveston  77554               TX   
92541             1     0        245  Galveston  77554               TX   
92542             0     0        287  Galveston  77554               TX   

          region_name_short  
0       Vail / Beaver Creek  
1       Vail / Beaver Creek  
2       Vail / Beaver Creek  
3       Vail / Beaver Creek  
4       Vail / Beaver Creek  
...                     ...  
92536  The Texas Gulf Coast  
92538  The Texas Gulf Coast  
92540  The Texas Gulf Coast  
92541  The Texas Gulf Coast  
92542  The Texas Gulf Coast  

[72456 rows x 14 columns]

In [6]:
# Get value counts for all unit_id numbers
df['unit_id'].value_counts()

9131     157
11033    146
6309     145
17653    143
40618    143
        ... 
31087      1
31237      1
9107       1
30653      1
17594      1
Name: unit_id, Length: 2125, dtype: int64

#### ^ I was curious what the rentals looked like without the ones that may have cancelled. Which, if the 0 amount paid is due to cancellation, then the answer would be 72,456 complete reservations total in the dataset & 2,125 units that were actually used. I am saving it to a csv, to see the difference in data. 


In [7]:
#renaming dataframe
zero_sales2 = df
zero_sales2

unit_id check_in_date check_out_date reservation_type  \
0        40630      1/1/2018       1/3/2018            guest   
1        40630      1/3/2018       1/7/2018            guest   
2        40630     1/11/2018      1/15/2018            guest   
3        40630     1/15/2018      1/18/2018            owner   
4        40630     1/20/2018      1/21/2018            guest   
...        ...           ...            ...              ...   
92536    21239      3/9/2019      3/13/2019            guest   
92538    21239     3/21/2019      3/24/2019            guest   
92540    21239      4/9/2019      4/12/2019            guest   
92541    21239     4/12/2019      4/14/2019            guest   
92542    21239     4/14/2019      4/16/2019            guest   

             reservation_source_type reservation_created_date  adult_guests  \
0                            unknown                 1/1/2018             2   
1                            unknown                 1/3/2018             2   
2                            unknown                9/29/2017             2   
3                            unknown                1/15/2018             2   
4                            unknown                1/17/2018             2   
...                              ...                      ...           ...   
92536  3rd Party Channel Reservation                2/17/2019             4   
92538  3rd Party Channel Reservation                3/20/2019             4   
92540  3rd Party Channel Reservation                 4/8/2019             2   
92541                         Vacasa                 4/8/2019             2   
92542  3rd Party Channel Reservation                 4/8/2019             6   

       child_guests  pets total_paid  city_name   zip5 state_name_short  \
0                 0     0       1930       Vail  81657               CO   
1                 0     0       2969       Vail  81657               CO   
2                 0     0      12913       Vail  81657               CO   
3                 0     0         60       Vail  81657               CO   
4                 0     0        515       Vail  81657               CO   
...             ...   ...        ...        ...    ...              ...   
92536             0     0       1300  Galveston  77554               TX   
92538             0     0        661  Galveston  77554               TX   
92540             0     0        340  Galveston  77554               TX   
92541             1     0        245  Galveston  77554               TX   
92542             0     0        287  Galveston  77554               TX   

          region_name_short  
0       Vail / Beaver Creek  
1       Vail / Beaver Creek  
2       Vail / Beaver Creek  
3       Vail / Beaver Creek  
4       Vail / Beaver Creek  
...                     ...  
92536  The Texas Gulf Coast  
92538  The Texas Gulf Coast  
92540  The Texas Gulf Coast  
92541  The Texas Gulf Coast  
92542  The Texas Gulf Coast  

[72456 rows x 14 columns]

In [8]:
# Save dataframe to csv 
zero_sales2.to_csv(r'cleaned_data2.csv')
zero_sales2

unit_id check_in_date check_out_date reservation_type  \
0        40630      1/1/2018       1/3/2018            guest   
1        40630      1/3/2018       1/7/2018            guest   
2        40630     1/11/2018      1/15/2018            guest   
3        40630     1/15/2018      1/18/2018            owner   
4        40630     1/20/2018      1/21/2018            guest   
...        ...           ...            ...              ...   
92536    21239      3/9/2019      3/13/2019            guest   
92538    21239     3/21/2019      3/24/2019            guest   
92540    21239      4/9/2019      4/12/2019            guest   
92541    21239     4/12/2019      4/14/2019            guest   
92542    21239     4/14/2019      4/16/2019            guest   

             reservation_source_type reservation_created_date  adult_guests  \
0                            unknown                 1/1/2018             2   
1                            unknown                 1/3/2018             2   
2                            unknown                9/29/2017             2   
3                            unknown                1/15/2018             2   
4                            unknown                1/17/2018             2   
...                              ...                      ...           ...   
92536  3rd Party Channel Reservation                2/17/2019             4   
92538  3rd Party Channel Reservation                3/20/2019             4   
92540  3rd Party Channel Reservation                 4/8/2019             2   
92541                         Vacasa                 4/8/2019             2   
92542  3rd Party Channel Reservation                 4/8/2019             6   

       child_guests  pets total_paid  city_name   zip5 state_name_short  \
0                 0     0       1930       Vail  81657               CO   
1                 0     0       2969       Vail  81657               CO   
2                 0     0      12913       Vail  81657               CO   
3                 0     0         60       Vail  81657               CO   
4                 0     0        515       Vail  81657               CO   
...             ...   ...        ...        ...    ...              ...   
92536             0     0       1300  Galveston  77554               TX   
92538             0     0        661  Galveston  77554               TX   
92540             0     0        340  Galveston  77554               TX   
92541             1     0        245  Galveston  77554               TX   
92542             0     0        287  Galveston  77554               TX   

          region_name_short  
0       Vail / Beaver Creek  
1       Vail / Beaver Creek  
2       Vail / Beaver Creek  
3       Vail / Beaver Creek  
4       Vail / Beaver Creek  
...                     ...  
92536  The Texas Gulf Coast  
92538  The Texas Gulf Coast  
92540  The Texas Gulf Coast  
92541  The Texas Gulf Coast  
92542  The Texas Gulf Coast  

[72456 rows x 14 columns]

#### Start of original uncleaned dataframe break down. 

In [9]:
# Preview original df again
original

unit_id check_in_date check_out_date reservation_type  \
0        40630      1/1/2018       1/3/2018            guest   
1        40630      1/3/2018       1/7/2018            guest   
2        40630     1/11/2018      1/15/2018            guest   
3        40630     1/15/2018      1/18/2018            owner   
4        40630     1/20/2018      1/21/2018            guest   
...        ...           ...            ...              ...   
92540    21239      4/9/2019      4/12/2019            guest   
92541    21239     4/12/2019      4/14/2019            guest   
92542    21239     4/14/2019      4/16/2019            guest   
92543    21239     4/16/2019      5/31/2019            owner   
92544    21239     5/31/2019     12/31/2021            owner   

             reservation_source_type reservation_created_date  adult_guests  \
0                            unknown                 1/1/2018             2   
1                            unknown                 1/3/2018             2   
2                            unknown                9/29/2017             2   
3                            unknown                1/15/2018             2   
4                            unknown                1/17/2018             2   
...                              ...                      ...           ...   
92540  3rd Party Channel Reservation                 4/8/2019             2   
92541                         Vacasa                 4/8/2019             2   
92542  3rd Party Channel Reservation                 4/8/2019             6   
92543                         Vacasa                4/13/2019             0   
92544                         Vacasa                4/13/2019             0   

       child_guests  pets total_paid  city_name   zip5 state_name_short  \
0                 0     0     $1,930       Vail  81657               CO   
1                 0     0     $2,969       Vail  81657               CO   
2                 0     0    $12,913       Vail  81657               CO   
3                 0     0        $60       Vail  81657               CO   
4                 0     0       $515       Vail  81657               CO   
...             ...   ...        ...        ...    ...              ...   
92540             0     0       $340  Galveston  77554               TX   
92541             1     0       $245  Galveston  77554               TX   
92542             0     0       $287  Galveston  77554               TX   
92543             0     0         $0  Galveston  77554               TX   
92544             0     0         $0  Galveston  77554               TX   

          region_name_short  
0       Vail / Beaver Creek  
1       Vail / Beaver Creek  
2       Vail / Beaver Creek  
3       Vail / Beaver Creek  
4       Vail / Beaver Creek  
...                     ...  
92540  The Texas Gulf Coast  
92541  The Texas Gulf Coast  
92542  The Texas Gulf Coast  
92543  The Texas Gulf Coast  
92544  The Texas Gulf Coast  

[92545 rows x 14 columns]

#### Question 3:

#### How many reservations in Vail?

Vail / Beaver Creek: 695

In [10]:
# Get count of reservations at Vail / Beaver Creek
# Reservations & Unit Countsd
original.groupby(by="region_name_short")["unit_id"].nunique()

region_name_short
Delaware                958
The Texas Gulf Coast    544
Vail / Beaver Creek     695
Name: unit_id, dtype: int64

#### Question 4:

#### How many check-in's were there in Vail in march of 2019?
Need assistance completing this in Pandas. 

In [11]:
# Empty

#### Question 5:

#### What % of reservations had children?

53.79%

In [12]:
original['child_guests'].mean()

0.5378680641849911

#### Question 6:

#### What's the average number of children on a reservation?
1.86 (almost 2)

In [13]:
# Get value count of # of children guests 
original['child_guests'].sum()

49777

In [14]:
# Divide # of reservations / # of children 
np.true_divide(92545, 49777)

1.8591919963035137

#### Question 7:

#### Complete table with Reservations & Unit Counts

   #### Reservations|  Units 
        
 Texas-     33,004  |  544
 
 Delaware-  29,830  |  958
 
 Vail-      29,7111  |  695

In [15]:
# Reservations & Unit Countsd
original.groupby(by="region_name_short")["unit_id"].nunique()

region_name_short
Delaware                958
The Texas Gulf Coast    544
Vail / Beaver Creek     695
Name: unit_id, dtype: int64

In [16]:
# Get count of reservations at Vail / Beaver Creek
original['region_name_short'].value_counts()

The Texas Gulf Coast    33004
Delaware                29830
Vail / Beaver Creek     29711
Name: region_name_short, dtype: int64

#### Question 8:

#### Total # of check-ins by month in each market

Need assistance completing this in Pandas.

In [31]:
#Empty

#### Question 9:

#### Calculate the average daily rate (ADR) for all reservations

Mean/Average = Getting an error, I believe it's because of 0's? I am not sure at the moment. Need more time to investigate. 

In [34]:
# Remove $ and , from total_paid
test['total_paid'] = test['total_paid'].str.replace('$', '')
test['total_paid'] = test['total_paid'].str.replace(',', '')
test

unit_id check_in_date check_out_date reservation_type  \
0        40630      1/1/2018       1/3/2018            guest   
1        40630      1/3/2018       1/7/2018            guest   
2        40630     1/11/2018      1/15/2018            guest   
3        40630     1/15/2018      1/18/2018            owner   
4        40630     1/20/2018      1/21/2018            guest   
...        ...           ...            ...              ...   
92540    21239      4/9/2019      4/12/2019            guest   
92541    21239     4/12/2019      4/14/2019            guest   
92542    21239     4/14/2019      4/16/2019            guest   
92543    21239     4/16/2019      5/31/2019            owner   
92544    21239     5/31/2019     12/31/2021            owner   

             reservation_source_type reservation_created_date  adult_guests  \
0                            unknown                 1/1/2018             2   
1                            unknown                 1/3/2018             2   
2                            unknown                9/29/2017             2   
3                            unknown                1/15/2018             2   
4                            unknown                1/17/2018             2   
...                              ...                      ...           ...   
92540  3rd Party Channel Reservation                 4/8/2019             2   
92541                         Vacasa                 4/8/2019             2   
92542  3rd Party Channel Reservation                 4/8/2019             6   
92543                         Vacasa                4/13/2019             0   
92544                         Vacasa                4/13/2019             0   

       child_guests  pets total_paid  city_name   zip5 state_name_short  \
0                 0     0       1930       Vail  81657               CO   
1                 0     0       2969       Vail  81657               CO   
2                 0     0      12913       Vail  81657               CO   
3                 0     0         60       Vail  81657               CO   
4                 0     0        515       Vail  81657               CO   
...             ...   ...        ...        ...    ...              ...   
92540             0     0        340  Galveston  77554               TX   
92541             1     0        245  Galveston  77554               TX   
92542             0     0        287  Galveston  77554               TX   
92543             0     0          0  Galveston  77554               TX   
92544             0     0          0  Galveston  77554               TX   

          region_name_short  
0       Vail / Beaver Creek  
1       Vail / Beaver Creek  
2       Vail / Beaver Creek  
3       Vail / Beaver Creek  
4       Vail / Beaver Creek  
...                     ...  
92540  The Texas Gulf Coast  
92541  The Texas Gulf Coast  
92542  The Texas Gulf Coast  
92543  The Texas Gulf Coast  
92544  The Texas Gulf Coast  

[92545 rows x 14 columns]

In [36]:
#renaming dataframe
test2 = test
test2

unit_id check_in_date check_out_date reservation_type  \
0        40630      1/1/2018       1/3/2018            guest   
1        40630      1/3/2018       1/7/2018            guest   
2        40630     1/11/2018      1/15/2018            guest   
3        40630     1/15/2018      1/18/2018            owner   
4        40630     1/20/2018      1/21/2018            guest   
...        ...           ...            ...              ...   
92540    21239      4/9/2019      4/12/2019            guest   
92541    21239     4/12/2019      4/14/2019            guest   
92542    21239     4/14/2019      4/16/2019            guest   
92543    21239     4/16/2019      5/31/2019            owner   
92544    21239     5/31/2019     12/31/2021            owner   

             reservation_source_type reservation_created_date  adult_guests  \
0                            unknown                 1/1/2018             2   
1                            unknown                 1/3/2018             2   
2                            unknown                9/29/2017             2   
3                            unknown                1/15/2018             2   
4                            unknown                1/17/2018             2   
...                              ...                      ...           ...   
92540  3rd Party Channel Reservation                 4/8/2019             2   
92541                         Vacasa                 4/8/2019             2   
92542  3rd Party Channel Reservation                 4/8/2019             6   
92543                         Vacasa                4/13/2019             0   
92544                         Vacasa                4/13/2019             0   

       child_guests  pets total_paid  city_name   zip5 state_name_short  \
0                 0     0       1930       Vail  81657               CO   
1                 0     0       2969       Vail  81657               CO   
2                 0     0      12913       Vail  81657               CO   
3                 0     0         60       Vail  81657               CO   
4                 0     0        515       Vail  81657               CO   
...             ...   ...        ...        ...    ...              ...   
92540             0     0        340  Galveston  77554               TX   
92541             1     0        245  Galveston  77554               TX   
92542             0     0        287  Galveston  77554               TX   
92543             0     0          0  Galveston  77554               TX   
92544             0     0          0  Galveston  77554               TX   

          region_name_short  
0       Vail / Beaver Creek  
1       Vail / Beaver Creek  
2       Vail / Beaver Creek  
3       Vail / Beaver Creek  
4       Vail / Beaver Creek  
...                     ...  
92540  The Texas Gulf Coast  
92541  The Texas Gulf Coast  
92542  The Texas Gulf Coast  
92543  The Texas Gulf Coast  
92544  The Texas Gulf Coast  

[92545 rows x 14 columns]

In [41]:
#test2['total_paid'].mean()

#### Question 10:

#### ADR per region
(Looking at Mean column)

Same thing as #9, there is an error that I can't figure out at this time, without digging further. 

In [42]:
original.groupby(by="region_name_short")["total_paid"].describe()

count unique top  freq
region_name_short                           
Delaware              29830   5116  $0  9457
The Texas Gulf Coast  33004   3089  $0  6653
Vail / Beaver Creek   29711   5742  $0  3979

#### Question 11:

Answered on cleaned_vacasa JPNB file. 